File: scripts/seed_huge.py
import json
import os
import urllib.request
import urllib.parse

def load_env():
    with open(".env.local", "r") as f:
        for line in f:
            if "=" in line:
                key, val = line.strip().split("=", 1)
                os.environ[key] = val.strip('"').strip("'")

load_env()
SUPABASE_URL = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
SUPABASE_KEY = os.environ.get("NEXT_PUBLIC_SUPABASE_ANON_KEY")

if not SUPABASE_URL or not SUPABASE_KEY:
    print("Missing Supabase credentials!")
    exit(1)

# 1. Clear existing database of old mutated duplicates
print("Wiping old database entries...")
del_req = urllib.request.Request(
    f"{SUPABASE_URL}/rest/v1/players?id=not.is.null",
    headers={
        "apikey": SUPABASE_KEY,
        "Authorization": f"Bearer {SUPABASE_KEY}",
        "Prefer": "return=minimal"
    },
    method="DELETE"
)
try:
    urllib.request.urlopen(del_req)
    print("Database wiped clean!")
except urllib.error.HTTPError as e:
    print(f"Delete step error: HTTP {e.code} - {e.read().decode('utf-8')}")
except Exception as e:
    print(f"Delete step note: {e}")

# 2. Upload deduplicated 1,329 roster
with open("Ipl players dataset/unique_players.json", "r", encoding="utf-8") as f:
    players = json.load(f)

print(f"Loaded {len(players)} UNIQUE mass players.")

headers = {
    "apikey": SUPABASE_KEY,
    "Authorization": f"Bearer {SUPABASE_KEY}",
    "Content-Type": "application/json",
    "Prefer": "resolution=merge-duplicates"
}

BATCH_SIZE = 300
inserted = 0

for i in range(0, len(players), BATCH_SIZE):
    chunk = players[i:i+BATCH_SIZE]
    req = urllib.request.Request(
        f"{SUPABASE_URL}/rest/v1/players",
        data=json.dumps(chunk).encode("utf-8"),
        headers=headers,
        method="POST"
    )
    
    try:
        urllib.request.urlopen(req)
        inserted += len(chunk)
        print(f"Batch {i//BATCH_SIZE + 1} completed! ({inserted}/{len(players)})")
    except urllib.error.HTTPError as e:
        print(f"Error on batch {i//BATCH_SIZE + 1}: HTTP {e.code} - {e.read().decode('utf-8')}")
    except Exception as e:
        print(f"Error on batch {i//BATCH_SIZE + 1}: {e}")

print(f"Successfully injected all {inserted} UNIQUE players into Supabase!")

File: scripts/seed-huge.ts
import fs from "fs";
import path from "path";
import dotenv from "dotenv";
import { createClient } from "@supabase/supabase-js";

dotenv.config({ path: path.resolve(process.cwd(), ".env.local") });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error("Missing Supabase credentials in .env.local");
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function main() {
  console.log("Starting MASSIVE Dataset Seeder...");

  const datasetPath = path.resolve(process.cwd(), "Ipl players dataset/3100_players.json");
  if (!fs.existsSync(datasetPath)) {
    console.error(`Dataset not found at ${datasetPath}`);
    process.exit(1);
  }

  const rawData = fs.readFileSync(datasetPath, "utf8");
  const players = JSON.parse(rawData);

  console.log(`Loaded dataset: ${players.length} historical players found.`);

  // Clear existing players to prevent strange mapping issues if they want a pure huge auction
  // Actually, upsert will naturally overwrite. Since the IDs change (e.g. P0606-2008), 
  // It won't overwrite the original 2026 ones (they were just P0606). It will just inject 3193 MORE.
  
  console.log("\nUpserting Players in ultra-batches of 500...");
  const BATCH_SIZE = 500;
  let playersInserted = 0;

  for (let i = 0; i < players.length; i += BATCH_SIZE) {
    const chunk = players.slice(i, i + BATCH_SIZE);
    
    const { error: playerError } = await supabase.from("players").upsert(chunk);

    if (playerError) {
      console.error(`❌ Error in batch ${i / BATCH_SIZE + 1}:`, playerError.message);
    } else {
      playersInserted += chunk.length;
      console.log(`✅ Batch ${i / BATCH_SIZE + 1} completed (${playersInserted}/${players.length} total)`);
    }
  }

  console.log("\n🎉 Database fully seeded successfully! You now have a massive auction pool.");
}

main().catch(console.error);
